zerver/models.py

```diff
@@ -329,6 +330,7 @@ class Realm(models.Model):
         video_chat_provider=int,
         waiting_period_threshold=int,
         digest_weekday=int,
+        only_admins_create_user_groups=bool,
```

Comment:
Let's implement this as a dropdown menu similar to `invite_to_stream_policy`; we may in the future want to have an intermediate "moderator" role that might be allowed to manage user groups.  I think a good name would be `user_group_edit_policy`, with values like these:

    USER_GROUP_EDIT_MEMBERS = 1 (default)
    USER_GROUP_EDIT_ADMINS = 2 (new option)

(And we can add more later as needed).